staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/mutating/dispatcher.go

```diff
@@ -56,9 +57,34 @@ func newMutatingDispatcher(p *Plugin) func(cm *webhook.ClientManager) generic.Di
 var _ generic.Dispatcher = &mutatingDispatcher{}
 
 func (a *mutatingDispatcher) Dispatch(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces, relevantHooks []*generic.WebhookInvocation) error {
+	reinvokeCtx := attr.GetReinvocationContext()
+	var webhookReinvokeCtx *webhookReinvokeContext
+	if v := reinvokeCtx.Value(PluginName); v != nil {
+		webhookReinvokeCtx = v.(*webhookReinvokeContext)
+	} else {
+		webhookReinvokeCtx = &webhookReinvokeContext{}
+		reinvokeCtx.SetValue(PluginName, webhookReinvokeCtx)
+	}
+
+	reinvokeUpToHere := func() {
+		webhookReinvokeCtx.RequireReinvokingPreviouslyInvokedPlugins()
+		reinvokeCtx.SetShouldReinvoke()
+	}
+
+	if reinvokeCtx.IsReinvoke() && webhookReinvokeCtx.IsOutputChangedSinceLastWebhookInvocation(attr.GetObject()) {
+		// If the object has changed, we know the in-tree plugin re-invocations have mutated the object,
+		// and we need to reinvoke all eligible webhooks.
+		reinvokeUpToHere()
```

Comment:
I was slightly confused by calling reinvokeUpToHere() outside the loop (especially since it calls `reinvokeCtx.SetShouldReinvoke()`, which I assume is a no-op today, but if we ever did allow more than one iteration, we wouldn't actually want to call `SetShouldReinvoke` here)

maybe just call `webhookReinvokeCtx.RequireReinvokingPreviouslyInvokedPlugins()` here, and inline `reinvokeUpToHere` below?

Reply:
Your right. It's noop now but wouldn't be compatible with 2+ iterations. Fixed.